components/brave_wallet_ui/common/trezor/trezor_bridge_keyring.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  TrezorDerivationPaths, TrezorBridgeAccountsPayload
+} from '../../components/desktop/popup-modals/add-account-modal/hardware-wallet-connect/types'
+import {
+  kTrezorHardwareVendor
+} from '../../constants/types'
+import {
+  TrezorCommand,
+  UnlockCommand,
+  GetAccountsCommand,
+  UnlockResponse,
+  GetAccountsResponsePayload,
+  TrezorAccount,
+
+  TrezorError,
+  TrezorBridgeTransport
+} from '../../common/trezor/trezor-messages'
+import { getLocale } from '../../../common/locale'
+
+export default class TrezorBridgeKeyring extends EventEmitter {
+  constructor () {
+    super()
+    this.unlocked_ = false
+    this.transport_ = new TrezorBridgeTransport()
+  }
+
+  type = () => {
+    return kTrezorHardwareVendor
+  }
+
+  getAccounts = async (from: number, to: number, scheme: string) => {
+    if (from < 0) {
+      from = 0
+    }
+    if (!this.isUnlocked() && !(await this.unlock())) {
+      return new Error(getLocale('braveWalletUnlockError'))
+    }
+    const paths = []
+    for (let i = from; i <= to; i++) {
+      paths.push(this.getPathForIndex(i, scheme))
+    }
+
+    const accounts = await this.getAccountsFromDevice(paths)
+    if (!accounts.success) {
+      throw Error(accounts.error)
+    }
+    return accounts.accounts
+  }
+
+  isUnlocked = () => {
+    return this.unlocked_
+  }
+
+  unlock = () => {
+    return new Promise(async (resolve, reject) => {
+      // @ts-ignore
+      const messageId: string = crypto.randomUUID()
+      this.getTransport().addEventListener(messageId, (data: UnlockResponse) => {
+        this.unlocked_ = data.result
+        if (data.result) {
+          resolve(true)
+        } else {
+          reject(false)
+        }
+      })
+      const message: UnlockCommand = { id: messageId, origin: window.origin,
+        command: TrezorCommand.Unlock }
+      await this.getTransport().postMessage(message)
+    })
+  }
+  getTransport = () => {
+    return this.transport_
+  }
+  private getAccountsFromDevice = async (paths: string[]): Promise<TrezorBridgeAccountsPayload> => {
+    return new Promise(async (resolve, reject) => {
+      const requestedPaths = []
+      for (const path of paths) {
+        requestedPaths.push({ path: path })
+      }
+      // @ts-ignore
+      const messageId: string = crypto.randomUUID()
+      this.getTransport().addEventListener(messageId, (data: GetAccountsResponsePayload) => {
+        if (data.payload.success) {
+          let accounts = []
+          for (const value of data.payload.payload as TrezorAccount[]) {
+            const buffer = Buffer.from(value.publicKey, 'hex')
+            const address = publicToAddress(buffer, true).toString('hex')
+            accounts.push({
+              address: toChecksumAddress(`0x${address}`),
+              derivationPath: value.serializedPath,
+              name: this.type(),
+              hardwareVendor: this.type(),
+              deviceId: String(value.fingerprint)
```

Comment:
this fingerprint is here is only first 4 bytes of full identifier and not collision free
```
Note that the fingerprint of the parent only serves as a fast way to detect parent and child nodes in software, and software must be willing to deal with collisions. Internally, the full 160-bit identifier could be used.
```
https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki

We should use same method of how we get Ledger's device id which is sha256 of first account address

Reply:
done